### ISSUE TITLE
Issue after update

### ISSUE THREAD

**Issue by lgertel** [fastify/fastify#998]
Hello everyone! My application just stopped working after update fastify!

```
> nomikos@1.0.0 dev /Users/lgertel/nomikos
> fastify start -l info app.js -P --inspect

/Users/lgertel/nomikos/node_modules/fastify/lib/decorate.js:12
  if (typeof fn.getter === 'function' || typeof fn.setter === 'function') {
                ^

TypeError: Cannot read property 'getter' of undefined
    at Object.decorate (/Users/lgertel/nomikos/node_modules/fastify/lib/decorate.js:12:17)
    at /Users/lgertel/nomikos/plugins/mongoose.js:9:13
    at Plugin.exec (/Users/lgertel/nomikos/node_modules/avvio/plugin.js:45:17)
    at Boot.loadPlugin (/Users/lgertel/nomikos/node_modules/avvio/plugin.js:125:10)
    at Task.release (/Users/lgertel/nomikos/node_modules/fastq/queue.js:127:16)
    at worked (/Users/lgertel/nomikos/node_modules/fastq/queue.js:169:10)
    at toLoad.finish (/Users/lgertel/nomikos/node_modules/avvio/plugin.js:128:7)
    at done (/Users/lgertel/nomikos/node_modules/avvio/plugin.js:86:5)
    at check (/Users/lgertel/nomikos/node_modules/avvio/plugin.js:97:7)
    at process._tickCallback (internal/process/next_tick.js:61:11)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nomikos@1.0.0 dev: `fastify start -l info app.js -P --inspect`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the nomikos@1.0.0 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lgertel/.npm/_logs/2018-06-26T15_53_17_993Z-debug.log
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by delvedor** [comment]
Can you paste some more code?
Especially the one you have at `at /Users/lgertel/nomikos/plugins/mongoose.js:9:13`.

**Comment by lgertel** [comment]
```
'use strict'

const fp = require('fastify-plugin')
const Mongoose = require('mongoose')

module.exports = fp(function (fastify, opts, next) {
  Mongoose.connect('mongodb://localhost/nomikos')
  if (!fastify.mongoose) {
    fastify.decorate('mongoose')
  }

  next()
})
```

**Comment by jsumners** [comment]
It's the `fastify.decorate('mongoose')` line. You are decorating with `undefined` which by definition will not have any properties.

Does it make sense to decorate the server with an `unrefined` value? If so, we need to patch this area of the code to verify `fn` is defined.

**Comment by delvedor** [comment]
@jsumners probably a truthy check can be useful, a user can add also `null` or a boolean value.

**Comment by delvedor** [comment]
Fixed in #999
